base_layer/wallet/src/text_message_service.rs

```diff
@@ -254,13 +270,44 @@ impl TextMessageService {
     /// Return a copy of the current lists of messages
     /// TODO Remove this in memory storing of message in favour of Sqlite persistence
     fn get_current_messages(&self) -> TextMessages {
+        let mut pending_messages: Vec<TextMessage> = Vec::new();
+
+        for (_k, v) in self.pending_messages.iter() {
```

Comment:
```rust
let pending_messages = self.pending_messages.iter().cloned().collect::<Vec<TextMessage>>();
```

But is there a difference between the for loop and simply cloning the vec as was deleted below?

Reply:
As below, self.pending_messages is a HashMap not a vector.